src/Npgsql.DependencyInjection/Npgsql.DependencyInjection.csproj

```diff
@@ -3,8 +3,8 @@
     <PropertyGroup>
         <Authors>Shay Rojansky</Authors>
         <!-- DbDataSource was introduced to .NET in net7.0. Before that Npgsql has its own built-in copy. -->
-        <TargetFrameworks>netstandard2.0;net7.0</TargetFrameworks>
-        <TargetFrameworks Condition="'$(DeveloperBuild)' == 'True'">net7.0</TargetFrameworks>
+        <TargetFrameworks>netstandard2.0;net7.0;net8.0</TargetFrameworks>
```

Comment:
Is there a reason this package needs to target net8.0 (like do we anticipate needing a specific .net 8.0 API)?

Reply:
No (at least not yet). I was assuming that we want to always include the LTS release for lifecycle reasons rather than for API reasons but that's definitely debatable.

Reply:
OK. I *think* net7.0 is there because we need something specific from there. I don't think we've preemptively added TLS TFMs without a specific reason so far, but I don't have very strong feelings here.

Reply:
Ok, I've investigated this and there are multiple reasons we have net7.0 here and IMHO none of them justifies why it is exactly 7.0.

1. We use the System.Diagnostics.CodeAnalysis shim
   This shim emulates APIs that were added for `netcoreapp3.0`/`netstandard2.1` and APIs that were added for `net5.0` so if we don't want to use our emulated APIs we'd need at least `net5.0`.

2. We use the following NuGet packages either directly or indirectly via Npgsql:

   - System.Threading.Channels 7.0.0
   - System.Text.Encodings.Web 7.0.0
   - System.Text.Json 7.0.2
   - System.Collections.Immutable 7.0.0
   - Microsoft.Extensions.DependencyInjection.Abstractions 7.0.0
   
   All of them raise TFM support build warnings if the TFM is less than `net6.0` so if we want to get rid of these warnings (and avoid the potential incompatibilities they warn about) we'd need at least `net6.0`

3. It looks like you cannot (reliably) override `TargetFrameworks` with `TargetFramework` which causes issues with $(DeveloperBuild)' == 'True'
   `TargetFrameworks` seems to take precedence over `TargetFramework` when both are active which leads to the fact that dependent libraries cannot resolve their dependencies for developer builds if `TargetFrameworks` does not contain the TFM specified for the developer build. Since we're moving developer builds to `net8.0` with this PR we'd need `net8.0`. But since our developer builds are a purely internal concept to make our lives easier I see this issue as a very bad justification to add or upgrade externally visible TFMs. That's why I suggest solving this differently.
   My suggested solution (add $(DeveloperBuild)' != 'True' to the default builds) will be part of this PR.

Reply:
PS: My suggestion would be to use 'netstandard2.0;net6.0' here and solve the DeveloperBuild issue as suggested above.

Reply:
PPS: To solve issue 2 we could also add a reference to the (very lightweight - no dependencies) `Microsoft.CodeAnalysis.PublicApiAnalyzers` package instead of using the shim like we do in some other packages (e.g. Npgsql.GeoJSON). TBH I don't see a good reason why we sometimes do it this way and sometimes that way.

Reply:
> My suggestion would be to use 'netstandard2.0;net6.0' here and solve the DeveloperBuild issue as suggested above.

As shown by the test failures we need net7.0 (because of DataSource), so 'netstandard2.0;net7.0' it will be...